src/sentry/api/endpoints/organization_seer_explorer_chat.py

```diff
@@ -0,0 +1,164 @@
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.models.organization import Organization
+from sentry.seer.seer_setup import get_seer_org_acknowledgement
+from sentry.seer.signed_seer_api import sign_with_seer_secret
+from sentry.types.ratelimit import RateLimit, RateLimitCategory
+
+
+def _call_seer_explorer_chat(
+    organization: Organization,
+    run_id: str | None,
+    query: str,
+    insert_index: int | None = None,
+    message_timestamp: float | None = None,
+):
+    """Call Seer explorer chat endpoint."""
+    path = "/v1/automation/explorer/chat"
+    body = orjson.dumps(
+        {
+            "organization_id": organization.id,
+            "run_id": run_id,
+            "query": query,
+            "insert_index": insert_index,
+            "message_timestamp": message_timestamp,
+        },
+        option=orjson.OPT_NON_STR_KEYS,
+    )
+
+    response = requests.post(
+        f"{settings.SEER_AUTOFIX_URL}{path}",
+        data=body,
+        headers={
+            "content-type": "application/json;charset=utf-8",
+            **sign_with_seer_secret(body),
+        },
+    )
+
+    response.raise_for_status()
+    return response.json()
+
+
+def _call_seer_explorer_state(organization: Organization, run_id: str):
+    """Call Seer explorer state endpoint."""
+    path = "/v1/automation/explorer/state"
+    body = orjson.dumps(
+        {
+            "run_id": run_id,
+            "organization_id": organization.id,
+        },
+        option=orjson.OPT_NON_STR_KEYS,
+    )
+
+    response = requests.post(
+        f"{settings.SEER_AUTOFIX_URL}{path}",
+        data=body,
+        headers={
+            "content-type": "application/json;charset=utf-8",
+            **sign_with_seer_secret(body),
+        },
+    )
+
+    response.raise_for_status()
+    return response.json()
+
+
+@region_silo_endpoint
+class OrganizationSeerExplorerChatEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ML_AI
+    enforce_rate_limit = True
+    rate_limits = {
+        "POST": {
+            RateLimitCategory.IP: RateLimit(limit=25, window=60),
+            RateLimitCategory.USER: RateLimit(limit=25, window=60),
+            RateLimitCategory.ORGANIZATION: RateLimit(limit=100, window=60 * 60),
+        },
+        "GET": {
+            RateLimitCategory.IP: RateLimit(limit=100, window=60),
+            RateLimitCategory.USER: RateLimit(limit=100, window=60),
+            RateLimitCategory.ORGANIZATION: RateLimit(limit=1000, window=60),
+        },
+    }
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        Get the current state of a Seer Explorer session.
+        """
+        if not request.user.is_authenticated:
+            return Response(status=400)
+
+        run_id = request.GET.get("run_id")
```

Comment:
consider adding `run_id` to be a url parameter instead of a query param, as it aligns more closely with RESTful API standards